File: 1916-find-center-of-star-graph/find-center-of-star-graph.py
class Solution(object):
    def findCenter(self, edges):
        n = len(edges) + 1
        graph = [[] for _ in range(n + 1)]

        for u, v in edges:
            graph[u].append(v)
            graph[v].append(u)
        
        for i in range(1, n + 1):
            if len(graph[i]) == n - 1:
                return i

File: 1916-find-center-of-star-graph/find-center-of-star-graph.java
class Solution {
    public int findCenter(int[][] edges) {
        int n = edges.length + 1;
        List<Integer>[] graph = new ArrayList[n + 1];
        for (int i = 0; i < n + 1; i++) {
            graph[i] = new ArrayList<>();
        }

        for (int[] edge : edges) {
            int u = edge[0];
            int v = edge[1];
            graph[u].add(v);
            graph[v].add(u);
        }

        for (int i = 1; i < n + 1; i++) {
            if (graph[i].size() == n - 1) {
                return i;
            }
        }

        return -1;
    }
}

